src/pal/src/file/directory.cpp

```diff
@@ -255,9 +263,13 @@ RemoveDirectoryW(
         dwLastError = ERROR_PATH_NOT_FOUND;
         goto done;
     }
-    
-    mb_size = WideCharToMultiByte( CP_ACP, 0, lpPathName, -1, mb_dir, MAX_LONGPATH,
+
+    length = (PAL_wcslen(lpPathName)+1) * sizeof(WCHAR);
```

Comment:
The size is not sufficient, valid WCHAR characters here can be encoded in upto 4 UTF-8 bytes (CP_ACP on Unix is UTF-8). WCHARs with code >= 0x800 and code < 0x10000 need 3 bytes and WCHARs with code >= 0x10000 need 4 bytes. But the 4 byte ones would be encoded as two WCHAR characters (using high / low surrogate), so we really need to multiply by 3 here to ensure we have a sufficient buffer.